src/wallet/rpcwallet.cpp

```diff
@@ -3041,6 +3041,49 @@ UniValue loadwallet(const JSONRPCRequest& request)
     return obj;
 }
 
+UniValue createwallet(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            "createwallet \"filename\"\n"
+            "\nCreates and loads a new wallet.\n"
+            "\nArguments:\n"
+            "1. \"wallet_name\"    (string, required) The name for the new wallet.\n"
+            "\nResult:\n"
+            "<walletname>     (string) The wallet name if created successfully.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("createwallet", "\"test.dat\"")
+            + HelpExampleRpc("createwallet", "\"test.dat\"")
+        );
+    std::string wallet_name = request.params[0].get_str();
+    std::string error;
+
+    fs::path wallet_path = fs::absolute(wallet_name, GetWalletDir());
+    if (fs::symlink_status(wallet_path).type() != fs::file_not_found) {
```

Comment:
This doesn't prevent concurrency which is very unlikely and so can be improved later.